llvm/lib/Analysis/ConstantFolding.cpp

```diff
@@ -3998,20 +3998,28 @@ ConstantFoldStructCall(StringRef Name, Intrinsic::ID IntrinsicID,
     if (!Vec)
       return nullptr;
 
-    unsigned NumElements =
-        cast<VectorType>(Vec->getType())->getElementCount().getKnownMinValue() /
-        2;
+    auto VecTy = cast<VectorType>(Vec->getType());
+    unsigned NumElements = VecTy->getElementCount().getKnownMinValue() / 2;
+
     SmallVector<Constant *, 4> Res0(NumElements), Res1(NumElements);
```

Comment:
Sink this SmallVector below the ConstantAggregateZero case so we don't unnecessarily create it.